### PR TITLE
Lua_Hyperspace_Module.md wiki update

### PR DESCRIPTION
**Pull request by THETASOLA**
Updating all entry in the wiki for exposed class/method/fields that were missing. And some cleaning.

ActivatedPowerDefinition::respawnCharges; was a duplicate if you were condering about that.

### CODE REVIEW

File: wiki/Lua-Hyperspace-Module.md [hunk 1]
```diff
@@ -1125,12 +1397,242 @@ function player_crew_health_boost()
...
+- `std::string :GetName()`
+- `bool :GetNewGoal()`
+- [`Point`](#Point) `:GetNextGoal()`
+- `int :GetRepairingId()`
+- `bool :GetResisted()`
+- [`Slot`](#Slot) `:GetSavedPosition()`
+- `int :GetSkillFromSystem(int systemId)`
+- `int :GetSkillLevel(int skillId)`
+- `float :GetSkillModifier(int skillId)`
+- [`int>`](#int>) `:GetSkillProgress(int skillId)`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1466]: Instances of `int>` should be `int`

File: wiki/Lua-Hyperspace-Module.md [hunk 2]
```diff
@@ -1125,12 +1397,242 @@ function player_crew_health_boost()
...
+- `int` `.width`
+- `int` `.height`
+- `pair<float, float> `.health`
+- `float` `.speed_x`
+- `float` `.speed_y`
+- [`Path`](#Path) `.path`
+- `bool` `.new_path`
+- `float` `.x_destination`
+- `float` `.y_destination`
+- [`Door*`](#Door) `.last_door`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1540]: Do we need to specify pointers in these docs? There's quite a few instances of these in this update

File: wiki/Lua-Hyperspace-Module.md [hunk 3]
```diff
@@ -1229,6 +1797,273 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+- `float` `.selfHealth`
+- `int` `.animFrame`
+- `bool` `.followCrew`
+- `bool` `.activateWhenReady`
+- `bool` `.activateReadyEnemies`
+- `std::string` `.transformRace`
+- [`std::vector<CrewSpawn*>`](#CrewSpawn) `.crewSpawns`
+- [`std::vector<StatBoostDefinition*>`](#StatBoostDefinition) `.statBoosts`
+- [`std::vector<StatBoostDefinition*>`](#StatBoostDefinition) `.roomStatBoosts`
+- `std::array<std::string,2>` `.event`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1933]: Should have a space after the comma

File: wiki/Lua-Hyperspace-Module.md [hunk 4]
```diff
@@ -1229,6 +1797,273 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+- [`ActivatedPowerRequirements*`](#ActivatedPowerRequirements) `.chargeReq`
+
+## ActivatedPowerRequirements
+
+### Fields
+- [`Type`](#Type) `.type`
+   - `int` `.PLAYER`
+   - `int` `.ENEMY`
+   - `int` `.CHARGE`
+   - `int` `.UNKNOWN`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1976]: Not sure about this method of documenting an enum

File: wiki/Lua-Hyperspace-Module.md [hunk 5]
```diff
@@ -1571,3 +2371,984 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+### Fields
+- `std::string` `.type`
+- `int` `.min`
+- `int` `.max`
+- `int` `.amount`
+- `bool` `.breach`
+
+## CustomShipUnlocks
+
+`Hyperspace.CustomShipUnlocks.instance`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:2431]: Preface this with "Accessed via"

File: wiki/Lua-Hyperspace-Module.md [hunk 6]
```diff
@@ -1571,3 +2371,984 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+- `int :GetType()`
+
+### Fields
+- `std::string` `.name`
+- [`Description`](#Description) `.desc`
+- `int` `.type`
+
+## ShipGraph
+
+Acquired via `Hyperspace.ShipGraph.GetShipInfo(int shipId)`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:2887]: Change "Acquired" to "Accessed"

File: wiki/Lua-Hyperspace-Module.md [hunk 7]
```diff
@@ -1571,3 +2371,984 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+- `float` `.lastWorldHeading`
+- [`Rect`](#Rect) `.shipBox`
+- `std::string` `.shipName`
+
+## ShieldPower
+
+### Fields
+- `int` `.first`
+- `int` `.second`
+- `pair<int, int>` `.super`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:2935]: I believe this should be an `std::pair`?

**Review by ChronoVortex — CHANGES_REQUESTED** [review]

File: wiki/Lua-Hyperspace-Module.md (hunk 2 above)
Comment (THETASOLA) [reply wiki/Lua-Hyperspace-Module.md:1540]: I feel like it's an important matter to know if we are dealing with a pointer to the class for modification purpose, like for the .choices thing that weren't giving back pointer and thus modification would have no effect.

File: wiki/Lua-Hyperspace-Module.md (hunk 4 above)
Comment (THETASOLA) [reply wiki/Lua-Hyperspace-Module.md:1976]: Frankly those enums thing were annoying to handle so if you have a better way I'm all ears

**Review by ChronoVortex — APPROVED** [review]

**Pull request merged**